README.md

```diff
@@ -1,229 +1,894 @@
-                             . find_and_alloc_map                      (kernel/bpf/syscall.c:122:8)
-           41us [-ENOMEM]    array_map_alloc+0x1eb                     (kernel/bpf/arraymap.c:150:16)
-                             . bpf_array_alloc_percpu                  (kernel/bpf/arraymap.c:39:6)
-           16us [NULL]       bpf_map_alloc_percpu+0x3f                 (kernel/bpf/syscall.c:436:2)
-                             . set_active_memcg                        (include/linux/sched/mm.h:315:6)
-                             . preempt_count                           (arch/x86/include/asm/preempt.h:27:9)
-           14us [NULL]       __alloc_percpu_gfp+0x5                    (mm/percpu.c:1894:9)
-            0us [NULL]       pcpu_alloc+0x5                            (mm/percpu.c:1679:1)
+20:19:36.372607 -> 20:19:36.372682 TID/PID 8346/8346 (simfail/simfail):
+
+                    entry_SYSCALL_64_after_hwframe+0x63  (arch/x86/entry/entry_64.S:120:0)
+                    do_syscall_64+0x35                   (arch/x86/entry/common.c:80:7)
+                    . do_syscall_x64                     (arch/x86/entry/common.c:50:12)
+    73us [-ENOMEM]  __x64_sys_bpf+0x1a                   (kernel/bpf/syscall.c:5067:1)
+    70us [-ENOMEM]  __sys_bpf+0x38b                      (kernel/bpf/syscall.c:4947:9)
+                    . map_create                         (kernel/bpf/syscall.c:1106:8)
+                    . find_and_alloc_map                 (kernel/bpf/syscall.c:132:5)
+!   50us [-ENOMEM]  array_map_alloc
+!*   2us [NULL]     bpf_map_alloc_percpu
+
+
+^C
+Detaching... DONE in 251 ms.
 ```
 
-Here three different error call stacks were captured and inline symbolization
-was performed. Functions with '. ' prefix are inlined functions, detected by
-`addr2line` (see below).
+Let's try to parse what `retsnoop` printed out.
 
-The same set of errors in verbose mode looks like this:
+First, each completed sample gets a single-line header:
+```
+20:19:36.372607 -> 20:19:36.372682 TID/PID 8346/8346 (simfail/simfail):
+```
+
+`20:19:36.372607 -> 20:19:36.372682` shows the timestamps of when
+`__x64_sys_bpf` (earliest activated entry function) was called and
+subsequently returned. `TID/PID 8346/8346 (simfail/simfail)` prints out thread
+and process IDs, as well as thread and process short names (so-called `comm`
+in kernel lingo, which you can get from `/proc/<pid>/comm`), respectively.
+This information lets you identify in the context of which process and thread
+stack trace was captured.
+
+After this we see stack trace with some additional information. Note that
+functions on the top are calling functions below them, forming a top-down call
+chain.
+
+Each line in a stack trace corresponds to one kernel function call. Kernel
+function name is in the center (e.g,. `__x64_sys_bpf`, `map_create`,
+`array_map_alloc`, etc), plus a hexadecimal offset (in bytes) within that
+function, which allows to pinpoint exactly assembly instruction that was
```

Comment:
s/which allows to pinpoint exactly assembly instruction/which allows the user to pinpoint the exact assembly instruction